drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -160,12 +160,13 @@ def calculate(&post_processor)
     private def disability?(enrollment, disability_type, disability_responses = [1])
       # Find most recent disability record associated with the enrollment for the appropriate type with a value
       # recorded before the end of the reporting period
-      disability = enrollment.disabilities.detect do |d|
+      disability = enrollment.disabilities.order(InformationDate: :desc).detect do |d|
         d.InformationDate < @filter.end_date &&
           d.DisabilityType == disability_type &&
           d.DisabilityResponse.in?([0, 1, 2, 3])
       end
-      disability.present? && disability.IndefiniteAndImpairs == 1 && disability.DisabilityResponse.in?(disability_responses)
+      impairing = [6] # developmental disability (4.06) does not require I&I
+      disability.present? && (disability.IndefiniteAndImpairs == 1 || disability_type.in?(impairing)) && disability.DisabilityResponse.in?(disability_responses)
```

Comment:
Instead of re-encoding the logic here, can we just lean on `GrdaWarehouse::Hud::Disability#indefinite_and_impairs?`?

8 also doesn't need I&I.

Reply:
Thanks! I didn't know that existed... I had left 8 out because it isn't in the report, but I agree it is safer.